sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -273,11 +319,8 @@ func (m *networkFlowManager) enrichConnections() {
 func (m *networkFlowManager) enrichAndSend() {
 	currentConns, currentEndpoints := m.currentEnrichedConnsAndEndpoints()
 
-	updatedConns := computeUpdatedConns(currentConns, m.enrichedConnsLastSentState)
-	updatedEndpoints := computeUpdatedEndpoints(currentEndpoints, m.enrichedEndpointsLastSentState)
-
-	m.enrichedConnsLastSentState = currentConns
-	m.enrichedEndpointsLastSentState = currentEndpoints
+	updatedConns := computeUpdatedConns(currentConns, m.enrichedConnsLastSentState, &m.lastSentStateMutex)
+	updatedEndpoints := computeUpdatedEndpoints(currentEndpoints, m.enrichedEndpointsLastSentState, &m.lastSentStateMutex)
```

Comment:
The only argument against that proposal is when one of the functions in the critical section panics, then we would never unlock the mutex. Not sure if that is a true risk...

Reply:
You can extract those for lines to another function and `defer`. It's a nit though.

Reply:
There is no problem in having these structs carry inconsistent data in case the connection breaks in-between each call. Also, the contention is not high enough to justify the performance improvement. Resolving.